runtime/js/40_http.js

```diff
@@ -35,10 +37,16 @@
           this.#rid,
         );
       } catch (error) {
+        // A connection error seen here would cause disrupted responses to throw
+        // a generic `BadResource` error. Instead store this error and replace
+        // those with it.
+        connErrors.set(this, error);
```

Comment:
If you're confident that this is indeed an Error object, then I think you could just save the message part of it since that's the only part that actually gets used down the line.

Reply:
The `.constructor` as well. Conceptually I'm cloning the error under a new call stack, it's simpler like this.

Reply:
Aaah, oh true yeah. I missed the `.constructor` usage. Yeah, saving the error object itself is good.